### ISSUE TITLE
[Bug]: Wrong placement of output files for ts_proto_library

### ISSUE THREAD

**Issue by jvliwanag** [aspect-build/rules_ts#466]
### What happened?

`ts_proto_library` should write its output on the directory the rule was defined.

Release 4.5.0 however writes the output instead on the directory where the proto file is.

### Version

Development (host) and target OS/architectures: macos

Output of `bazel --version`: aspect 5.7.2

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file: 4.5.0

Language(s) and/or frameworks involved: rules_ts

### How to reproduce

```shell
Create a ts_proto_library where the proto file is on a different directory + BUILD.

See sample [commit](https://github.com/aspect-build/rules_ts/commit/625990fb85164fc623ede643203eda6309f38d4e) from [sample branch](https://github.com/jvliwanag/rules_ts/tree/proto-ts-path-issue).
```

**Comment by alexeagle** [comment]
Yeah that looks like a simple pathing bug in computing the output path.
